core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java

```diff
@@ -1,28 +1,119 @@
 package org.stellar.anchor.sep38;
 
-import java.util.List;
-import org.stellar.anchor.asset.AssetInfo;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
 import org.stellar.anchor.asset.AssetService;
 import org.stellar.anchor.config.Sep38Config;
+import org.stellar.anchor.dto.sep38.GetPricesResponse;
 import org.stellar.anchor.dto.sep38.InfoResponse;
+import org.stellar.anchor.exception.AnchorException;
+import org.stellar.anchor.exception.BadRequestException;
+import org.stellar.anchor.exception.NotFoundException;
+import org.stellar.anchor.exception.ServerErrorException;
+import org.stellar.anchor.integration.rate.GetRateRequest;
+import org.stellar.anchor.integration.rate.GetRateResponse;
 import org.stellar.anchor.integration.rate.RateIntegration;
 import org.stellar.anchor.util.Log;
 
 public class Sep38Service {
   final Sep38Config sep38Config;
   final AssetService assetService;
   final RateIntegration rateIntegration;
+  final InfoResponse infoResponse;
+  final Map<String, InfoResponse.Asset> assetMap;
 
   public Sep38Service(
       Sep38Config sep38Config, AssetService assetService, RateIntegration rateIntegration) {
     this.sep38Config = sep38Config;
     this.assetService = assetService;
     this.rateIntegration = rateIntegration;
+    this.infoResponse = new InfoResponse(this.assetService.listAllAssets());
+    assetMap = new HashMap<>();
+    this.infoResponse.getAssets().forEach(asset -> assetMap.put(asset.getAsset(), asset));
     Log.info("Initializing sep38 service.");
   }
 
   public InfoResponse getInfo() {
-    List<AssetInfo> assets = this.assetService.listAllAssets();
-    return new InfoResponse(assets);
+    return this.infoResponse;
+  }
+
+  public GetPricesResponse getPrices(
+      String sellAssetName,
+      String sellAmount,
+      String countryCode,
+      String sellDeliveryMethod,
+      String buyDeliveryMethod)
+      throws AnchorException {
+    if (this.rateIntegration == null) {
+      throw new ServerErrorException("internal server error");
+    }
+    validateGetPricesInput(
+        sellAssetName, sellAmount, countryCode, sellDeliveryMethod, buyDeliveryMethod);
+
+    InfoResponse.Asset sellAsset = assetMap.get(sellAssetName);
+    assert sellAsset != null;
```

Comment:
❓ How is an assertion error handled in the callers of this function? Shouldn't the behavior in this case be similar to the behavior that we when `this.rateIntegration` is null?